Examples/src/main/kotlin/com/mapbox/vision/examples/UsbVideoSourceActivityKt.kt

```diff
@@ -0,0 +1,229 @@
+        setContentView(R.layout.activity_main)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        startVisionManager()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        stopVisionManager()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        vision_view.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        vision_view.onPause()
+    }
+
+    private fun startVisionManager() {
+        if (allPermissionsGranted() && !visionManagerWasInit) {
+            VisionManager.create(usbVideoSource)
+            VisionManager.setModelPerformanceConfig(
+                ModelPerformanceConfig.Merged(
+                    ModelPerformance.On(ModelPerformanceMode.FIXED, ModelPerformanceRate.HIGH)
+                )
+            )
+            vision_view.setVisionManager(VisionManager)
+            VisionManager.visionEventsListener = visionEventsListener
+            VisionManager.start()
+
+            visionManagerWasInit = true
+        }
+    }
+
+    private fun stopVisionManager() {
+        if (visionManagerWasInit) {
+            VisionManager.stop()
+            VisionManager.destroy()
+
+            visionManagerWasInit = false
+        }
+    }
+
+    private val onDeviceConnectListener: USBMonitor.OnDeviceConnectListener =
+        object : USBMonitor.OnDeviceConnectListener {
+            override fun onAttach(device: UsbDevice?) {
+                toast("Usb camera attached")
+                usbMonitor?.requestPermission(device!!)
+            }
+
+            override fun onConnect(
+                device: UsbDevice?,
+                ctrlBlock: USBMonitor.UsbControlBlock?,
+                createNew: Boolean
+            ) {
+                toast("Usb camera connected")
+                backgroundHandler.post {
+                    synchronized(this@UsbVideoSourceActivityKt) {
+                        uvcCamera?.destroy()
+                        uvcCamera = initializeUvcCamera(ctrlBlock!!)
+                    }
+                }
+            }
+
+            override fun onDetach(device: UsbDevice?) {
+                toast("Usb camera detached")
+            }
+
+            override fun onCancel(p0: UsbDevice?) {
+                toast("Usb camera cancelled")
+            }
+
+            override fun onDisconnect(device: UsbDevice?, ctrlBlock: USBMonitor.UsbControlBlock?) {
+                toast("Usb camera disconnected")
+                backgroundHandler.post {
+                    uvcCamera?.close()
+                }
+            }
+        }
+
+    private fun initializeUvcCamera(ctrlBlock: USBMonitor.UsbControlBlock) = UVCCamera().also { camera ->
+        camera.open(ctrlBlock)
+        camera.setPreviewSize(
+            IMAGE_SIZE.imageWidth,
+            IMAGE_SIZE.imageHeight,
+            UVCCamera.FRAME_FORMAT_MJPEG
+        )
+
+        val surfaceTexture = SurfaceTexture(createExternalGlTexture())
```

Comment:
nit. SurfaceTexture(createExternalGlTexture()).apply {
      setDefaultBufferSize(
            IMAGE_SIZE.imageWidth,
            IMAGE_SIZE.imageHeight
        )
}

Reply:
Don't like it here, reduces only extra 1 line while adding lambda eg. complexity and verbosity